snuba/datasets/entities/events.py

```diff
@@ -103,8 +104,23 @@ def callback_func(
     query: Query,
     request_settings: RequestSettings,
     referrer: str,
+    query_metadata: SnubaQueryMetadata,
     results: List[Result[QueryResult]],
 ) -> None:
+    cache_hit = False
+    is_duplicate = False
+
+    # Captures if any of the queries involved was a cache hit or duplicate, as cache
+    # hits may a cause of inconsistency between results.
+    # Doesn't attempt to distinguish between all of the specific scenarios (one or both
+    # queries, or splits of those queries could have hit the cache).
+    if any([query.stats.get("cache_hit", 0) for query in query_metadata.query_list]):
+        cache_hit = True
+    elif any(
+        [query.stats.get("is_duplicate", 0) for query in query_metadata.query_list]
+    ):
+        is_duplicate = True
```

Comment:
QueryResult contains the stats accumulated during the execution. You should not need to pass metadata back and forth
https://github.com/getsentry/snuba/blob/master/snuba/web/__init__.py#L8-L28

Reply:
omg I'm dumb 😭